gui/mass-remove.lua

```diff
@@ -0,0 +1,336 @@
+        dfhack.buildings.deconstruct(building)
+    end
+end
+
+function MassRemoveUI:unremoveBuilding(x, y, z)
+    local building = dfhack.buildings.findAtTile(x, y, z)
+    if building then
+        for i, job in ipairs(building.jobs) do
+            if job.job_type == df.job_type.DestroyBuilding then
+                building.jobs:erase(i)
+                break
+            end
+        end
+    end
+end
+
+function MassRemoveUI:changeDesignation(x, y, z)
+    if self.action == "suspend" then
+        self:suspend(x, y, z)
+    elseif self.action == "unsuspend" then
+        self:unsuspend(x, y, z)
+    elseif self.action == "remove_x" then
+        self:removeBuilding(x, y, z)
+    elseif self.action == "unremove_x" then
+        self:unremoveBuilding(x, y, z)
+    elseif self.action == "remove_n" then
+        self:removeConstruction(x, y, z)
+    elseif self.action == "unremove_n" then
+        self:unremoveConstruction(x, y, z)
+    elseif self.action == "remove_a" then
+        self:removeBuilding(x, y, z)
+        self:removeConstruction(x, y, z)
+    elseif self.action == "unremove_a" then
+        self:unremoveBuilding(x, y, z)
+        self:unremoveConstruction(x, y, z)
+    end
+end
+
+function MassRemoveUI:changeDesignations(x1, y1, z1, x2, y2, z2)
+    local x_start, x_end = minToMax(x1, x2)
+    local y_start, y_end = minToMax(y1, y2)
+    local z_start, z_end = minToMax(z1, z2)
+    for x=x_start, x_end do
+        for y=y_start, y_end do
+            for z=z_start, z_end do
+                self:changeDesignation(x, y, z)
+            end
+        end
+    end
+end
+
+function MassRemoveUI:getColor(action)
+    if action == self.action then
+        return COLOR_WHITE
+    else
+        return COLOR_GREY
+    end
+end
+
+function MassRemoveUI:renderOverlay()
+    local vp=self:getViewport()
+    local dc = gui.Painter.new(self.df_layout.map)
+
+    --show buildings/constructions marked for removal and planned buildings/constructions that are suspended
+    if gui.blink_visible(500) then
+        local joblist = df.global.world.jobs.list.next
+        while joblist do
+            local job = joblist.item
+            joblist = joblist.next
+
+            if job.job_type == df.job_type.ConstructBuilding and job.flags.suspend then
+                paintMapTile(dc, vp, nil, job.pos, "s", COLOR_LIGHTRED)
+            elseif job.job_type == df.job_type.RemoveConstruction then
+                paintMapTile(dc, vp, nil, job.pos, "n", COLOR_LIGHTRED)
+            end
+        end
+
+        for x=vp.x1, vp.x2 do
+            for y=vp.y1, vp.y2 do
+                local building = dfhack.buildings.findAtTile(x, y, vp.z)
+                if building and dfhack.buildings.markedForRemoval(building) then
+                    paintMapTile(dc, vp, nil, xyz2pos(x, y, vp.z), "x", COLOR_LIGHTRED)
+                end
+            end
+        end
+    end
+
+    --show box selection
+    if gui.blink_visible(250) and self.marking then
```

Comment:
We don't have a standard here, but I would probably increase this to `400` or `500` since it's potentially a large blinking area.

Reply:
I mainly want the box selection blinking so that you can see what is under it, so you know what you're removing. So if I match the `500` blink delay of the "marked buildings/constructions" overlay, it'll be hard to see if the buildings/constructions you're selecting are marked or not.

After playing with it for a bit, I decided to go for `if not gui.blink_visible(250)`, so that the two blinks alternate and you can see both the box selection and the marks. Give it a try and let me know if it looks ok!

Implemented in 439612e